core/src/com/etheller/warsmash/viewer5/handlers/w3x/simulation/behaviors/build/CBehaviorHumanRepair.java

```diff
@@ -42,17 +44,66 @@ public boolean isWithinRange(CSimulation simulation) {
 
     @Override
     protected CBehavior update(CSimulation simulation, boolean withinFacingWindow) {
-        unit.getUnitAnimationListener().playAnimation(false, AnimationTokens.PrimaryTag.STAND,
-                SequenceUtils.WORK, 1.0f, true);
         if(this.target instanceof CWidget) {
             CWidget targetWidget = (CWidget) this.target;
-            float newLifeValue = targetWidget.getLife() + 1;
-            boolean done = newLifeValue > targetWidget.getMaxLife();
-            if(done) {
-                newLifeValue = targetWidget.getMaxLife();
-            }
-            targetWidget.setLife(simulation, newLifeValue);
-            if(done) {
+            //progress construction here (MFROMAZ)
+            if(targetWidget.getClass()==CUnit.class) {
+                CUnit targetUnit = ((CUnit) targetWidget);
+                final CPlayer player = simulation.getPlayer(unit.getPlayerIndex());
+                float newLifeValue = targetWidget.getLife() +
+                        ((WarsmashConstants.SIMULATION_STEP_TIME / (targetUnit.getUnitType().getBuildTime()*this.ability.getRepairTimeRatio()))
+                                * (targetUnit.getMaxLife()));
+                float costs_gold = ((WarsmashConstants.SIMULATION_STEP_TIME / (targetUnit.getUnitType().getBuildTime()*this.ability.getRepairTimeRatio()))
```

Comment:
Everywhere else in the codebase I used Java variable naming convention -- i.e. camel case -- instead of underscores. It's fairly pedantic and if this was my only concern I'd probably still merge this anyway, but if you find yourself reviewing this it could be nice for consistency to use `costsGold` instead of `costs_gold`.